lib/services/local/users.go

```diff
@@ -670,8 +679,30 @@ func (s *IdentityService) DeleteUser(ctx context.Context, user string) error {
 	// each user has multiple related entries in the backend,
 	// so use DeleteRange to make sure we get them all
 	startKey := backend.ExactKey(webPrefix, usersPrefix, user)
-	err = s.DeleteRange(ctx, startKey, backend.RangeEnd(startKey))
-	return trace.Wrap(err)
+	if err = s.DeleteRange(ctx, startKey, backend.RangeEnd(startKey)); err != nil {
+		return trace.Wrap(err)
+	}
+
+	// Delete notification objects associated with this user.
+	if (s.notificationsSvc != &NotificationsService{}) {
```

Comment:
This is just a precautionary check to prevent a panic in case the notifications service wasn't properly initialized. I can't think of any scenario where this would happen, but if it does I'd rather the notifications cleanup not run than a panic